blobstore-fs/src/main.rs

```diff
@@ -509,11 +544,13 @@ impl Blobstore for FsProvider {
         let mut errors = Vec::new();
 
         for object in &arg.objects {
-            let opath = Path::join(&Path::join(&root, &arg.container_id), &object);
-            if let Err(e) = remove_file(opath.as_path()) {
+            let object_subpath = Path::new(&arg.container_id).join(&object);
+            let object_path = self.resolve_subpath(&root, &object_subpath)?;
```

Comment:
Nit: you should just be able to pass `object_subpath` without a borrow. This goes for everywhere in this file